lib/components/app/print-layout.tsx

```diff
@@ -54,14 +57,26 @@ class PrintLayout extends Component<Props, State> {
   }
 
   componentDidMount() {
-    const { itinerary, location, parseUrlQueryString } = this.props
+    const {
+      currentQuery,
+      itinerary,
+      location,
+      parseUrlQueryString,
+      setMapCenter
+    } = this.props
+    // TODO: this is an annoying hack. Ideally we wouldn't wipe out initLat and initLon
+    // TODO: is there a way to adjust transitiveData to force the transitive overlay to re-adjust bounds?
+    const { lat, lon } = currentQuery.from
+
     // Add print-view class to html tag to ensure that iOS scroll fix only applies
     // to non-print views.
     addPrintViewClassToRootHtml()
     // Parse the URL query parameters, if present
     if (!itinerary && location && location.search) {
       parseUrlQueryString()
     }
+
+    setMapCenter({ lat, lon })
```

Comment:
To change the map center dynamically, the preferred way is to use [`map.jumpTo`](https://maplibre.org/maplibre-gl-js-docs/api/map/#map#jumpto). Access the `map` instance by adding a [`<MapProvider>`](https://visgl.github.io/react-map-gl/docs/api-reference/map-provider), probably in `ResponsiveWebApp`, and using `this.context` according to [this doc](https://reactjs.org/docs/context.html#classcontexttype).

Reply:
Yes this is how I originally intended to do this, however complexities of how and where we render the map mean that this is easier said than done. As I mentioned in my comment below, I will use a workaround I found where we force the overlays to update by updating their inputs